File: projects/code_combat/4_Backwoods_Forest/095-Backwoods_Ambush/back_ambush.py
# You need an Elemental codex 1+ to cast "Haste"

def attackIfExists(x, y):
    hero.moveXY(x, y)
    enemy = hero.findNearestEnemy()
    if enemy:
        while enemy.health > 0:
            hero.attack(enemy)

hero.cast("haste", hero)

attackIfExists(24, 42)
attackIfExists(27, 60)
attackIfExists(42, 50)
attackIfExists(39, 24)

File: projects/code_combat/4_Backwoods_Forest/095-Backwoods_Ambush/backAmbush.js
hero.moveXY(24, 42);
var enemy = hero.findNearestEnemy();
if(enemy) {
    hero.attack(enemy);
    hero.attack(enemy);
}

hero.moveXY(27, 60);
enemy = hero.findNearestEnemy();
if(enemy) {
    // Attack the enemy if it exists!
    hero.attack(enemy);
    hero.attack(enemy);
}

hero.moveXY(42, 50);
enemy = hero.findNearestEnemy();
// Use an if-statement to check if an enemy exists.
if (enemy) {
    // Attack the enemy if it exists!
    hero.attack(enemy);
    hero.attack(enemy);
}

hero.moveXY(39, 24);
// Find the nearest enemy:
enemy = hero.findNearestEnemy();

// Check if the enemy exists:
if (enemy) {
    // Attack the enemy if it exists!
    hero.attack(enemy);
    hero.attack(enemy);
}
